src/Entity/OgMembership.php

```diff
@@ -320,6 +320,22 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
     }
 
+    if (!$this->get('entity_id')->value) {
+      // Group was not set.
+      throw new \LogicException('Membership cannot be set for an empty or an unsaved group.');
```

Comment:
> I think you may have to explicitly call the validate method for that to run.

Yes, that's what I know about it as-well.

So, the flow should be add validation and then call them via `preSave()` ourself? It sounds a bit wrong to me.